src/lib/swapper/swappers/ArbitrumBridgeSwapper/utils/fetchArbitrumBridgeSwap.ts

```diff
@@ -0,0 +1,63 @@
+import { Erc20Bridger, EthBridger, getL2Network } from '@arbitrum/sdk'
+import type {
+  L1ToL2TransactionRequest,
+  L2ToL1TransactionRequest,
+} from '@arbitrum/sdk/dist/lib/dataEntities/transactionRequest'
+import { ethAssetId, ethChainId, fromAssetId } from '@shapeshiftoss/caip'
+import { isEvmChainId } from '@shapeshiftoss/chain-adapters'
+import { type Asset, KnownChainIds } from '@shapeshiftoss/types'
+import { BigNumber } from 'ethers5'
+import { arbitrum } from 'viem/chains'
+import { getEthersV5Provider } from 'lib/ethersProviderSingleton'
+
+export type FetchArbitrumBridgeSwapInput = {
+  affiliateBps: string
+  buyAsset: Asset
+  receiveAddress: string
+  sellAmountIncludingProtocolFeesCryptoBaseUnit: string
+  sellAsset: Asset
+  maximumSlippageDecimalPercentage: string
+  sendAddress: string
+}
+
+export const fetchArbitrumBridgeSwap = async ({
+  buyAsset,
+  sellAmountIncludingProtocolFeesCryptoBaseUnit,
+  sellAsset,
+  sendAddress,
+  receiveAddress,
+}: FetchArbitrumBridgeSwapInput): Promise<
+  Omit<L1ToL2TransactionRequest | L2ToL1TransactionRequest, 'retryableData'>
+> => {
+  const l2Network = await getL2Network(arbitrum.id)
+  const isDeposit = sellAsset.chainId === ethChainId
+  const isEthBridge = isDeposit ? sellAsset.assetId === ethAssetId : buyAsset.assetId === ethAssetId
+
+  const bridger = isEthBridge ? new EthBridger(l2Network) : new Erc20Bridger(l2Network)
+
+  if (!(isEvmChainId(sellAsset.chainId) && isEvmChainId(buyAsset.chainId))) {
+    throw new Error(`Arbitrum Bridge only supports EVM chains`)
+  }
+
+  const erc20L1Address = fromAssetId((isDeposit ? sellAsset : buyAsset).assetId).assetReference
+  const l1Provider = getEthersV5Provider(KnownChainIds.EthereumMainnet)
+  const l2Provider = getEthersV5Provider(KnownChainIds.ArbitrumMainnet)
+
+  const request = await (isDeposit
+    ? bridger.getDepositRequest({
```

Comment:
deposit is missing the concept of destination address currently if we want to support manual receive address, though I am not sure why it doesn't exist on the eth bridge. If not supported for eth deposit, we will want to make sure that isn't shown in the trade ui (haven't looked yet)

Reply:
This is precisely the reason why I've disabled crossAccountTrade for the sake of simplicity for now!
With that being said, we should still have a `destinationAddress` for deposits regardless.
b1e229b170